keith-vscode/src/pasta/stpa-interaction.ts

```
@@ -0,0 +1,133 @@
+                    handleWorkSpaceEdit(currentUri.toString(), formulas, new vscode.Position(0, 0));
+                }
+            }
+            quickPick.hide();
+        });
+        quickPick.onDidHide(() => quickPick.dispose());
+        quickPick.show();
+    }
+}
+
+/**
+ * An SCChart LTL annotation.
+ * @param formula The formula for the annoation.
+ * @param description The description for the annotation,
+ * @returns the SCChart LTL annotation for the given arguments.
+ */
+const ltlAnnotation = (formula: string, description: string): string => {
+    return "@LTL \"" + formula + "\", \"" + description + "\" \n";
+};
+
+/**
+ * Applys a workspaceedit to the document defined by {@code uri}, at the position {@code position} with the given {@code text}.
+ * @param uri URI of the document where the edit should be applied.
+ * @param text The text to insert in the document.
+ * @param position The position in the document where the text should be inserted.
+ */
+async function handleWorkSpaceEdit(uri: string, text: string, position: vscode.Position): Promise<void> {
+    // get the desired editor and document
+    const editor = vscode.window.visibleTextEditors.find(visibleEditor => visibleEditor.document.uri.toString() === uri);
+    const textDocument = editor?.document;
+    if (!textDocument) {
+        console.error(
+            `Server requested a text edit but the requested uri was not found among the known documents: ${uri}`
+        );
+        return;
+    }
+    // create the insert workspaceedit
+    const workSpaceEdit = new vscode.WorkspaceEdit();
+    const edits: vscode.TextEdit[] = [vscode.TextEdit.insert(position, text)];
+    workSpaceEdit.set(textDocument.uri, edits);
+
+    // Apply and save the edit. Report possible failures.
+    const edited = await vscode.workspace.applyEdit(workSpaceEdit);
+    if (!edited) {
+        console.error("Workspace edit could not be applied!");
```

can you report an error to the user instead of a console log